src/mat.js

```diff
@@ -55,38 +55,44 @@
      * @return matrix(i0:i1,j0:j1)
      */
     self.webgazer.mat.getSubMatrix = function(matrix, i0, i1, j0, j1){
-        var X = new Array(i1-i0+1);
-        for(var i=0; i<i1-i0+1; i++){
-            X[i] = new Array(j1-j0+1);
-        }
+        var size = i1-i0+1,
```

Comment:
I may be misreading this, but shouldn't this be (j1 - j0 + 1) not (i1 - i0 + 1)?